src/Microsoft.AspNet.Mvc.Core/ParameterBinding/ModelBindingHelper.cs

```diff
@@ -381,5 +387,54 @@ public static string CreatePropertyModelName(string prefix, string propertyName)
                 return prefix + "." + propertyName;
             }
         }
+
+        internal static void ClearModelStateDictionaryForModel(
```

Comment:
Would like to see at least a unit test for each of the cases that matter here. Functional tests are good but they are much harder to debug and write, we shouldn't rely on them solely.